content/docs/developers/_index.md

```diff
@@ -0,0 +1,145 @@
++++
+title = "Developers"
+weight = 300
+[extra]
+emoji = "⚒️"
+tile = "I want to develop a client, bridge or bot"
+last_updated = "2022-10-18T16:00:00Z"
++++
+
+## Elements of Matrix
+
+### Homeserver
+
+A homeserver is a piece of software hosting accounts of Matrix users. It is
+bound to a single domain that cannot change over time. The accounts on a server
+have an identifier made of a local part, and a server part. A typical identifier
+would be
+
+```txt
+@username:example.com
+```
+
+Users on a server can send `events` into `rooms`. An event is a particular json
+object, describing what a user is trying to do (join a room, send a message,
+update a specific value…). In the case of instant messaging, rooms are very
+similar to Slack, Discord or IRC channels. Most of the events in such rooms are
+about sending messages. Rooms have a unique technical identifier, and zero or
+several human-readable aliases or addresses. Aliases are made of a room name,
+and a server part. A typical room alias would be:
+
+```
+#mountain-bike:example.com
+```
+
+Homeservers are federated: the Matrix specification defines a [Sever-Server API](https://spec.matrix.org/latest/server-server-api/)
+(also known as Federation API) to describe interactions between servers.
+Whenever a user is in a room, their homeserver needs to have a local copy of
+that room.
+
+For example, if `@alice:outdoors.com` is the first user from `outdoors.com` to
+try to join `#mountain-bike:example.com`, then her homeserver is going to reach
+out to `example.com` to get a copy of the room. `outdoors.com` and `example.com`
+then stay in touch to synchronise their copy of the room.
+
+Whenever the homeserver receives new events, it's in charge of parsing them,
+perform checks on the event, and take action accordingly (e.g. add a specific 
+user to a specific room when it receives a room join event). The expected
```

Comment:
Maybe it's just me, but what does "add a specific user to a specific room" mean if not sending a join event? Maybe we could just write something like "e.g. sending messages from users on the homeserver to other participating homeservers, or distributing messages from other participating homeservers to users"?